Directory.Build.props

```diff
@@ -1,7 +1,7 @@
 <Project>
   <PropertyGroup>
     <IsPackable>false</IsPackable>
-    <LangVersion>12</LangVersion>
+    <LangVersion>13</LangVersion>
```

Comment:
Is the change using any new language features that make this necessary? I wouldn't object to bumping however C# 12 is the minimum suported baseline that the source generator is targeting, so using that at least for the test types project is a good way to extensive coverage for the source generator.

Reply:
I presume it's to support `params` span? I think this would render the API unusable in C# 12. Given that it shouldn't be a particularly performance sensitive API I think sticking with param array should be fine.

Reply:
Yes, it was for the `params` span parameter. And that's fine, I'll switch to an array. 
Strictly speaking though, we might want to (still) copy the array we receive, since the caller can pass in an array and then have mutable access to the same object we use later, leading to undefined behavior. But whether we make that defensive copy should be based on the policy you hold for the library in general. Do you tend to write that off as "you shot your own foot" or do you make the copy? I'll go either way.

Reply:
Defensive copy sounds good.

Reply:
FWIW, our approach in the BCL has been to expose both span and array overloads, the idea being that each language version picks the appropriate overload. I wouldn't oppose doing it that way either, but that being said this shouldn't be a hot-path API.

Reply:
The lack of a reference to the immutable collections in the netstandard2.0 build of your library keeps surprising me. Is that an intentional choice? If not, I'd tend to use an `ImmutableArray` here, and I could safely expose that directly from the property, allowing for alloc-free enumeration for public users.